src/hotspot/share/opto/output.cpp

```diff
@@ -1362,8 +1362,10 @@ CodeBuffer* PhaseOutput::init_buffer() {
 
   int pad_req   = NativeCall::byte_size();
 
+  // Despite the name "stub", GC barrier stubs are emitted into
+  // the insn section, and should be counted in code_req.
   BarrierSetC2* bs = BarrierSet::barrier_set()->barrier_set_c2();
-  stub_req += bs->estimate_stub_size();
+  code_req += bs->estimate_stub_size();
```

Comment:
I would be careful with this. I tried this change as part of [JDK-8331253](https://git.openjdk.org/jdk/commit/3383ad6397d5a2d8fb232ffd3e29a54e0b37b686) fix. But then I have to rollback that to avoid regression [JDK-8333226](https://github.com/openjdk/jdk/pull/19531)

It is affecting inlining decision (InlineSmallCode) and have to be handled carefully.

Reply:
Here is PR for original change: https://github.com/openjdk/jdk/pull/19029

Reply:
On other hand this line change affects only requested size and not how final instructions section size is calculated. So it could be fine.

Reply:
Dang. I have not realized this is part of inline heuristics. Let me see if I can avoid touching this part.

Reply:
Yeah, this affects inlining heuristics in a major way, like the issues you have linked described. Additionally, I realized the whole `estimate_stub_size` does not actually work well: https://github.com/openjdk/jdk/pull/23333#issuecomment-2624140301. So I am making a lateral move here, and fixing C2 stub barrier expansion code directly for Leyden.